src/3D/setup3d.F90

```diff
@@ -906,16 +919,20 @@ subroutine setup(fileName, fileType)
         do i = 1, nUnique
             if (fullTopoType(i) == topoEdge .and. fullBCType(1, i) == BCDefault) then
                 print *, 'There was a missing boundary condition for edge node:', uniquePts(:, i)
-                stop
-
+                bcTopoError = .True.
             else if (fullTopoType(i) == topoCorner .and. (fullBCType(1, i) == BCDefault .or. &
                                                           fullBCType(2, i) == BCDefault)) then
                 print *, 'There was a missing boundary condition for corner node:', uniquePts(:, i)
-                stop
+                bcTopoError = .True.
             end if
-
         end do
 
+        ! if we ran into topology or BC errors, stop
+        if (bcTopoError) then
+            print *, 'pyHyp exited. See above errors.'
```

Comment:
Can we improve the error message a bit, either add visual cues like `ERROR:` or mention it exited prematurely or improperly, or both.